### PR TITLE
[GH-13459] Add unit tests to the "ArchiveChannelsCmd" mmctl command

### PR DESCRIPTION
**Pull request by anidok**
Summary
Add unit tests to the "ArchiveChannelsCmd" mmctl command

Ticket Link
Fixes [#13459](https://github.com/mattermost/mattermost-server/issues/13459)

Jira Link
https://mattermost.atlassian.net/browse/MM-21386

### CODE REVIEW

File: commands/channel.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ func addUserToChannel(c client.Client, channel *model.Channel, user *model.User,
 
 func archiveChannelsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	if len(args) < 1 {
-		return errors.New("Enter at least one channel to archive.")
+		return errors.New("Enter at least one channel to archive")
```
Comment (anidok) [inline commands/channel.go]: Note that the error message has been deliberately modified, otherwise golint throws the following warning.
```
error strings should not be capitalized or end with punctuation or a newline go-lint
```

File: commands/channel_test.go [hunk 2]
```diff
@@ -133,6 +136,277 @@ func (s *MmctlUnitTestSuite) TestSearchChannelCmdF() {
...
+		printer.Clean()
+
+		teamArg := "some-team-id"
+		mockTeam := model.Team{Id: teamArg}
+		channelArg := "some-channel"
+		channelID := "some-channel-id"
+		mockChannel := model.Channel{Id: channelID, Name: channelArg}
+
+		cmd := &cobra.Command{}
+		args := fmt.Sprintf("%s:%s", teamArg, channelArg)
```
Comment (mgdelacroix) [inline commands/channel_test.go]: To simplify the code here, we can directly use:

```suggestion
		args := teamArg + ":" + channelArg
```

File: commands/channel_test.go [hunk 3]
```diff
@@ -133,6 +136,277 @@ func (s *MmctlUnitTestSuite) TestSearchChannelCmdF() {
...
+
+	s.Run("Fail to archive a non-existing channel on an existent team", func() {
+		printer.Clean()
+
+		teamArg := "some-non-existing-team-id"
+		mockTeam := model.Team{Id: teamArg}
+		channelArg := "some-non-existing-channel"
+
+		cmd := &cobra.Command{}
+		args := []string{fmt.Sprintf("%s:%s", teamArg, channelArg)}
```
Comment (mgdelacroix) [inline commands/channel_test.go]:
```suggestion
		args := []string{teamArg + ":" + channelArg}
```

File: commands/channel_test.go [hunk 4]
```diff
@@ -133,6 +136,277 @@ func (s *MmctlUnitTestSuite) TestSearchChannelCmdF() {
...
+	})
+
+	s.Run("Fail to archive a channel on a non-existent team", func() {
+		printer.Clean()
+
+		teamArg := "some-non-existent-team-id"
+		channelArg := "some-channel"
+
+		cmd := &cobra.Command{}
+		args := []string{fmt.Sprintf("%s:%s", teamArg, channelArg)}
```
Comment (mgdelacroix) [inline commands/channel_test.go]:
```suggestion
		args := []string{teamArg + ":" + channelArg}
```

File: commands/channel_test.go [hunk 5]
```diff
@@ -133,6 +136,277 @@ func (s *MmctlUnitTestSuite) TestSearchChannelCmdF() {
...
+		err := archiveChannelsCmdF(s.client, cmd, args)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+
+		expected := printer.GetErrorLines()[0]
+		actual := fmt.Sprintf("Unable to find channel '%s'", args[0])
+		s.Require().Equal(expected, actual)
+	})
+
```
Comment (mgdelacroix) [inline commands/channel_test.go]: nit: unnecessary whitespace

**Comment by anidok** [comment]
Thanks @mgdelacroix for your feedback. I have accomodated the relevant changes requested by you. Please have a look at it and let me know if it looks good. :)

**Comment by anidok** [comment]
> LGTM @anidok, thanks for the changes!!

Thanks @mgdelacroix for your patience and valueable feedbacks.

**Comment by anidok** [comment]
I have resolved the conflicts. Let me know if it looks good to be merged now.

**Comment by grundleborg** [comment]
Thanks @anidok - looks good now.
